### ISSUE TITLE
No loss compile

### ISSUE THREAD

**Issue by lemuriandezapada** [fchollet/keras#1426]
I often find myself training a complex graph, then transferring some of the weight matrices to a smaller, simpler one, and use it to attach outputs and compute stuff. However compiling the graph with a loss function I'm never going to use when all I want to do is compute it in a forward manner takes too long for comfort.

Would there be any way to compile a network with no loss function just to run it in "production" mode?

**Comment by nouiz** [comment]
Just in case this is useful, this is old Pylearn2 code that allow to delay
the compilation of a Theano function to the first call to it:

https://github.com/lisa-lab/pylearn2/blob/master/pylearn2/utils/compile.py

**Comment by fchollet** [comment]
> However compiling the graph with a loss function I'm never going to use when all I want to do is compute it in a forward manner takes too long for comfort.

If you only need the function that maps the input of the network to its output, you can generate it manually without the Keras compile step:

```python
import keras.backend as K
# assuming Sequential model -- adapt it for Graph
m_input = model.get_input(train=False)
m_output = model.get_output(train=False)
f = K.function([m_input], [m_output])
res = f([x])[0]
```
